spec/models/subscription_spec.rb

```diff
@@ -66,4 +66,27 @@
       end
     end
   end
+
+  describe ".charged_at_least_once" do
+    let(:canceled_subscription) { create(:subscription, :canceled) }
+    let(:pending_payment_subscription) { create(:subscription, :pending_payment) }
+    let(:paid_subscription) { create(:subscription, :paid) }
+    let(:unpaid_subscription) { create(:subscription, :unpaid) }
+    let(:charged_subscriptions) do
+      [
+        canceled_subscription,
+        pending_payment_subscription,
+        paid_subscription,
+        unpaid_subscription
+      ]
+    end
+
+    before do
+      create(:subscription, :waiting_for_charging_day)
+    end
+
+    it "should return all subscriptions with :canceled, :pending_payment, :paid, :unpaid statuses" do
+      expect(Subscription.charged_at_least_once).to match_array charged_subscriptions
```

Comment:
You could have a `let` of a subscription that shouldn't be on the results to ensure its not adding non desired subscriptions.